Task7.c

```diff
@@ -0,0 +1,50 @@
+#include <stdio.h>
+#include <stdlib.h>
+
+int main()
+{
+    printf("Enter n to get prime numbers <= n: ");
+    int n = 0;
+    int scanned = scanf("%d", &n);
+    while (scanned == 0)
+    {
+        scanf("%*[^\n]");
+        printf("You didn't enter a number, try again: ");
+        scanned = scanf("%d", &n);
+    }
+    if (n > 1)
+    {
+        int* sieve = calloc(n + 1, sizeof(int));
```

Comment:
```suggestion
        bool* sieve = calloc(n + 1, sizeof(int));
```
Массив хранит значения вида "да" или "нет", поэтому `bool`.